subsys/bluetooth/audio/mcc.c

```diff
@@ -1049,6 +1032,14 @@ static uint8_t mcs_notify_handler(struct bt_conn *conn,
 	return BT_GATT_ITER_CONTINUE;
 }
 
+static void discovery_failure(struct bt_conn *conn, int err)
```

Comment:
Could just check the error code, and reset the instance if not zero.

Reply:
Do we not need to reset the MCS instance on discover complete? 

If we don't then we won't be able to call `bt_mcc_discover_mcs` again since it check for the pointer:

```c
int bt_mcc_discover_mcs(struct bt_conn *conn, bool subscribe)
{
	CHECKIF(!conn) {
		return -EINVAL;
	} else if (cur_mcs_inst) {
		return -EBUSY;
	}

```

Reply:
Changed.
Would like to postpone the reset functionality considerations, as it out of scope for this PR.